clio-server/src/main/scala/org/broadinstitute/clio/server/service/IndexService.scala

```diff
@@ -34,19 +39,46 @@ abstract class IndexService[CI <: ClioIndex](
 
   def upsertMetadata(
     indexKey: clioIndex.KeyType,
-    metadata: clioIndex.MetadataType
-  ): Future[UpsertId] = {
-    val updatedMetadata = metadata.withDocumentStatus(
-      metadata.documentStatus.orElse(Some(DocumentStatus.Normal))
-    )
-
-    persistenceService
-      .upsertMetadata(
-        indexKey,
-        updatedMetadata,
-        documentConverter,
-        elasticsearchIndex
-      )
+    metadata: clioIndex.MetadataType,
+    force: Boolean = false
+  )(implicit materializer: Materializer): Future[UpsertId] = {
+    // query by key to see if the document already exists
+    queryMetadataForKey(indexKey)
+      .runFold[Either[Exception, Json]](
```

Comment:
If we change the output type of `upsertMetadata` to `Source[UpsertId, NotUsed]` then we could do `fold` instead of `runFold` here, and have this all be one big stream. That should let us drop the implicit materializer too.